frontend/src/metabase/dashboard/actions/cards.js

```diff
@@ -85,22 +86,69 @@ export const addTextDashCardToDashboard = function ({ dashId }) {
   });
 };
 
-export const addActionDashCardToDashboard = ({ dashId }) => {
-  const virtualActionsCard = {
-    ...createCard(),
-    display: "action",
-    archived: false,
-  };
-  const dashcardOverrides = {
-    card: virtualActionsCard,
-    size_x: 2,
-    size_y: 1,
-    visualization_settings: {
-      virtual_card: virtualActionsCard,
-    },
+const esitmateCardSize = (displayType, action) => {
+  if (displayType === "button") {
+    return { size_x: 2, size_y: 1 };
+  }
+
+  return {
+    size_x: 6,
+    size_y: Math.round(3 + action.parameters.length * 1.5),
```

Comment:
Mind extracting `3` and `1.5` into constants here? I assume `1.5` is height per form field and `3` gives some space for the form title and footer, but it might not be obvious